blockchain/utxobackend.go

```diff
@@ -370,22 +516,12 @@ func (l *LevelDbUtxoBackend) dbFetchUtxoStats(dbTx database.Tx) (*UtxoStats, err
 
 // FetchStats returns statistics on the current UTXO set.
 func (l *LevelDbUtxoBackend) FetchStats() (*UtxoStats, error) {
-	var stats *UtxoStats
-	err := l.db.View(func(dbTx database.Tx) error {
-		var err error
-		stats, err = l.dbFetchUtxoStats(dbTx)
-		return err
-	})
-	if err != nil {
-		return nil, err
-	}
-
-	return stats, nil
+	return l.dbFetchUtxoStats()
```

Comment:
No reason, it was just following the same pattern as the other methods where the logic that actually accesses the database lives in the internal function.  I don't feel strongly about it either way, though, so updated as suggested.

Reply:
Well, the primary reasons the other ones did/do it that way are:

- When they require a database transaction which means it has to wrapped in a `db.Update/View` func
- When an overall mutex is required so that it's safe for outside callers, but can still be called internally without taking the mutex for cases when the internal code already holds it